### ISSUE TITLE
[inputs.systemd_units] Error in plugin: listing loaded units failed: dbus: connection closed by user

### ISSUE THREAD

**Issue by MironJajtic** [influxdata/telegraf#15111]
### Relevant telegraf.conf

```toml
[[inputs.systemd_units]]
  pattern = "auditd.service cagefs.service cdp-agent.service chronyd.service clamav-freshclam.service clamd.service cpanel_php_fpm.service cpanel.service crond.service csf.service dnsadmin.service dovecot.service dovecot.service exim.service haproxy.service httpd.service imunify*.service kcare.service lfd.service lsapi-cache.service lshttpd.service mariadb.service memcached*.service monit.service mysqld.service named.service NetworkManager.service nginx.service nis-domainname.service nscd.service php-fpm*.service plesk-*.service postfix.service puppet.service pure-ftpd.service qemu-guest-agent.service redis.service rhnsd.service rsyslog.service sbm-agent.service spamassassin.service spamd.service sshd.service tailwatchd.service telegraf.service tuned.service zabbix-agent.service"
  # unittype = "service"
  # details = false
  loaded_only = true
  timeout = "10s"
```

### Logs from Telegraf

```text
2024-04-05T10:28:15+02:00 E! [inputs.systemd_units] Error in plugin: listing loaded units failed: dbus: connection closed by user
2024-04-05T10:28:30+02:00 E! [inputs.systemd_units] Error in plugin: listing loaded units failed: dbus: connection closed by user
```

### System info

Telegraf 1.31.0-ee0025ec (git: pull/15108@ee0025ec), systemd 252 (252-18.el9), CloudLinux 9.3

### Steps to reproduce

1. Start telegraf with the [inputs.systemd_units] plugin enabled, everything is fine.
2. Restart dbus-broker.service 
3. Telegraf starts throwing errors "Error in plugin: listing loaded units failed: dbus: connection closed by user"
4. Restart telegraf.service, everything is fine again.

### Expected behavior

The expected behavior is that Telegraf does not start throwing errors endlessly when dbus-broker.service is restarted.

### Actual behavior

Telegraf started filling log file with errors:
```
2024-04-05T10:28:30+02:00 E! [inputs.systemd_units] Error in plugin: listing loaded units failed: dbus: connection closed by user
```

### Additional info

We noticed this started happening on Cloudlinux v9.3 with Telegraf v1.30.0, v1.30.1 and the same is with PR v1.31.0
We had no such errors with Telegraf v1.29.5.

**Comment by MironJajtic** [comment]
@srebhan thanks a lot, problem is solved with PR #15129

**Comment by srebhan** [comment]
@MironJajtic please keep the issue open, it will be automatically closed as soon as the corresponding PR is merged. This way, people will find the issue and we know what the current state is. Thanks for testing!
